inst/include/cpp11/doubles.hpp

```diff
@@ -134,33 +134,44 @@ typedef r_vector<double> doubles;
 
 }  // namespace writable
 
+template <>
+inline double na() {
+  return NA_REAL;
+}
+
+template <>
+inline bool is_na(const double& x) {
+  return ISNA(x);
+}
```

Comment:
Because we use `na<double>()` in `as_doubles()` below now, I had to either forward declare `template <> double na();` or move these up a little. I chose the latter since it seemed simpler and is already what we do in `integers.hpp`.